src/components/forms/JurisdictionSelect/index.tsx

```diff
@@ -46,14 +46,39 @@ export interface JurisdictionSelectProps<T = SelectOption> extends AsyncSelectPr
 export const promiseOptions = (
   service: OpenSRPService,
   paramsToUse: Dictionary,
-  hierarchy: SelectOption[]
+  hierarchy: SelectOption[],
+  jurisdictionStatus: boolean,
+  setFinalLocation: (value: boolean) => void,
+  setJurisdictionParam: (value: boolean) => void
 ) =>
   // tslint:disable-next-line:no-inferred-empty-object-type
-  new Promise((resolve, reject) =>
+  new Promise((resolve, reject) => {
     service
-      .list(paramsToUse)
-      .then((e: JurisdictionOption[]) => {
-        const options = e.map(item => {
+      .list({ ...paramsToUse, is_jurisdiction: jurisdictionStatus })
+      .then((jurisdictionLocationOptions: JurisdictionOption[]) => {
+        /** Check if payload has no name property then use id instead
+         *  If there is no location return no options
+         */
+        if (!jurisdictionStatus) {
+          setFinalLocation(true);
+          if (
+            jurisdictionLocationOptions.length >= 1 &&
+            !jurisdictionStatus &&
+            !jurisdictionLocationOptions[0].properties.name
+          ) {
+            const optionsWithoutName = jurisdictionLocationOptions.map(item => {
+              return { label: item.id, value: item.id };
+            });
```

Comment:
Do we need `optionsWithoutName` and `optionsWithoutName`, can we not change this `return { label: item.id, value: item.id }` to use name if available otherwise use id?  I think we can => idk if the syntax is correct but something like `label: item.name ? item.name : item.id`